src/order/preorder_hom.lean

```diff
@@ -118,43 +256,48 @@ instance {β : Type*} [order_bot β] : order_bot (α →ₘ β) :=
 
 @[simps]
 instance {β : Type*} [order_top β] : has_top (α →ₘ β) :=
-{ top := ⟨λ a, ⊤, λ a b h, le_refl _⟩ }
+{ top := const α ⊤ }
 
 instance {β : Type*} [order_top β] : order_top (α →ₘ β) :=
 { top := ⊤,
   le_top := λ a x, le_top,
   .. (_ : partial_order (α →ₘ β)) }
 
-@[simps]
 instance {β : Type*} [complete_lattice β] : has_Inf (α →ₘ β) :=
-{ Inf := λ s, ⟨ λ x, Inf ((λ f : _ →ₘ _, f x) '' s), λ x y h,
-    Inf_le_Inf_of_forall_exists_le begin
-      simp only [and_imp, exists_prop, set.mem_image, exists_exists_and_eq_and, exists_imp_distrib],
-      intros,
-      subst_vars,
-      refine ⟨_,by assumption, monotone _ h⟩
-    end ⟩ }
+{ Inf := λ s, ⟨λ x, ⨅ f ∈ s, (f : _) x, λ x y h, binfi_le_binfi (λ f _, f.mono h)⟩ }
```

Comment:
What's the advantage of the new definition over the old one? Naively, I'd go for the one that defines `Inf` in terms of `Inf` instead of defining `Inf` in terms of `infi`.

`pi.has_Inf` chooses to use `⨅ f : s` instead of `⨅ f ∈ s`, but I also don't know the tradeoff there.

Reply:
`pi.has_Inf` uses the same definition for `complete_lattice`s and `conditionally_complete_lattice`s. Here we can use a more readable definition and have shorter proofs.